### ISSUE TITLE
Navigating to file:// hangs Marionette

### ISSUE THREAD

**Issue by Buzz2Buzz** [mozilla/geckodriver#469]
## Meta -
OS:  
Windows 7
Selenium Version:  
3.0.1
Browser:  
Firefox

Browser Version:  
51.0.1 (Portable)

## Expected Behavior -
Open browser and go to a specific page

## Actual Behavior -
Stuck on user profile page

## Steps to reproduce -
http://stackoverflow.com/q/42154400/4675568

I'm executing this code:  
```
public class Selenium {

        public static void main(String[] args) {
            WebDriver driver = createFFDriver();
            driver.navigate().to("http:....");
            System.out.println("Finished");
        }

        public static WebDriver createFFDriver(){
            System.setProperty(GeckoDriverService.GECKO_DRIVER_EXE_PROPERTY,SELENIUM_DRIVERS + "\\firefox\\foo\\geckodriver.exe");
            FirefoxProfile firefoxProfile = new FirefoxProfile();
            File pathToBinary = new File(SELENIUM_INSTALLS_DEV + "\\51.0.1\\FirefoxPortable.exe");
            FirefoxBinary ffBinary = new FirefoxBinary(pathToBinary);

            return new FirefoxDriver(ffBinary,firefoxProfile);
        }
}
```
In logs, I've got this:  
```
1486713046153   geckodriver INFO    Listening on 127.0.0.1:12466 
Feb 10, 2017 8:50:46 AM org.openqa.selenium.remote.ProtocolHandshake
createSession INFO: Attempting bi-dialect session, assuming Postel's Law holds true on the remote end
1486713046731   mozprofile::profile INFO    Using profile path foo\AppData\Local\Temp\rust_mozprofile.p25D0Gb1sBQm
1486713046752   geckodriver::marionette INFO    Starting browser foo\firefox\51.0\FirefoxPortable.exe
1486713046782   geckodriver::marionette INFO    Connecting to Marionette on localhost:52818
```

Browser is opened but stuck.
![ff_profile_stuck](https://cloud.githubusercontent.com/assets/6714374/22880732/9d603aaa-f1e3-11e6-95f8-1e8cd29ffc9c.png)

**Comment by andreastt** [comment]
This only happens if you navigate to a file:// protocol: https://bugzilla.mozilla.org/show_bug.cgi?id=1332122

**Comment by Buzz2Buzz** [comment]
Oh, okay. Thanks for editing my post.
But I've got a question then, why the browsers opens up with this page specifically ? I didn't ask for it.

**Comment by andreastt** [comment]
It looks like you are appending the profile directory to the browser binary, thus causing Firefox to navigate to that URL.

**Comment by Buzz2Buzz** [comment]
I can make it work well with some changes:
- Using RemoteDriver with Capabilities, and a distant VM (this should be the production use case):
```
public static WebDriver createFFRemoteDriver() throws MalformedURLException {
        DesiredCapabilities capabilities = DesiredCapabilities.firefox();
        capabilities.setCapability("platform", "VISTA");
        capabilities.setCapability("version", "51.0.1");
        capabilities.setCapability("marionette", true);
        capabilities.setCapability("acceptSslCerts", true);
        return new RemoteWebDriver(new URL(REMOTE_HUB_URL),capabilities);
    }
```
- Using a standard Firefox. If I do the same with a Portable Edition, I face the same previous issue.

**Comment by andreastt** [comment]
“Portable Edition”, whatever it is, is not supported by Mozilla. I would recommend using an official build of Firefox. A guess from my side is that “Portable Edition” does not support profiles?  When it receives the `--profile <path to profile>` argument, it interprets it as a website address to navigate to.

**Comment by Buzz2Buzz** [comment]
There must be changes between old Firefox driver and Gecko driver then. With the old one, I used to do this way (with capabilities, and no specific profile). Maybe a default profile is used and then, the interpretation is done like you said.  
I'm going to investigate this, thanks for pointing me on this direction.  
For the moment I cannot drop Portable Edition but I take a note about this.

**Comment by andreastt** [comment]
Let me reiterate that Mozilla does not support Portable Edition and we cannot guarantee that it will function correctly with geckodriver or Marionette. It is required for us to start with a custom profile so that we can populate some preferences needed to interact with Firefox in automated environments.

**Comment by Buzz2Buzz** [comment]
Understood. For the moment we don't use Geckodriver but when we will, Portable Edition will be dropped as you suggest.

**Comment by vgrigoruk** [comment]
Same issue ( when when you're trying to load a static resource via http(s), for example the following url hangs marionette :
```
https://www.google.com/favicon.ico
```

**Comment by andreastt** [comment]
@vgrigoruk That should be fixed in Firefox 52 onwards: https://bugzilla.mozilla.org/show_bug.cgi?id=1322772

Please file a separate issue if it isn’t, but navigating to an ImageDocument is _not_ the same problem as this.

**Comment by andreastt** [comment]
This bug is tracked by https://bugzilla.mozilla.org/show_bug.cgi?id=1332122.

**Comment by whimboo** [comment]
And it is fixed in today's Nightly.

**Comment by akostadinov** [comment]
fixed on July 7th, I'm using GeckoDriver 0.18.0 which was released 11th July. But I still hit this `file://` issue.
Or is it fixed only when running FF56 (I'm running 55)?

**Comment by andreastt** [comment]
@akostadinov It is fixed in Firefox 56.
